crates/accelerate/src/circuit_library/utils.rs

```diff
@@ -0,0 +1,26 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use qiskit_circuit::{
+    operations::{Param, StandardGate},
+    packed_instruction::PackedOperation,
+    Clbit, Qubit,
+};
+use smallvec::SmallVec;
+
+pub(super) type StandardInstruction = (StandardGate, SmallVec<[Param; 3]>, SmallVec<[Qubit; 2]>);
+pub(super) type Instruction = (
+    PackedOperation,
+    SmallVec<[Param; 3]>,
+    Vec<Qubit>,
+    Vec<Clbit>,
+);
```

Comment:
I like this very much! Do you think it would make sense / be possible to also use these definition from within the `synthesis` directory? Or should we move these to an even more central place?

Reply:
Yeah I think that would make sense, though we might want to discuss the name a bit if we move this to a central location, so we don't mistake it among all the other instructions 😄

Reply:
As per the comment below, all the Rust changes are reverted in 3e009a5e7